cord19-explorer/src/main/java/edu/asu/diging/cord19/explorer/core/mongo/impl/SearchProviderRegistry.java

```diff
@@ -0,0 +1,30 @@
+package edu.asu.diging.cord19.explorer.core.mongo.impl;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import edu.asu.diging.cord19.explorer.web.model.SearchType;
+
+@Configuration
+public class SearchProviderRegistry {
+
+    @Autowired
+    private List<SearchProvider> searchProviders;
+    
+    
+    @Bean
+    public Map<SearchType, SearchProvider> initialize(){
+        Map<SearchType, SearchProvider> map = new HashMap<>();
+        for (SearchProvider search : searchProviders) {
+            map.put(search.getSearchType(), search);
+        }
+        return map;
```

Comment:
Your provider registry should be responsible for managing your search provider. This class should be a service, with a get method and whatever else you need to get specific search providers. By making the map a bean and anyone can change it as they like, you open the code up too many posssible side effects. What if a class just empties the map? Your registry should be in charge of all actions so you have one place to control all providers. Does this make sense?